conda_build/deprecations.py

```diff
@@ -18,6 +18,8 @@
 
 from . import __version__
 
+warnings.filterwarnings("default", category=DeprecationWarning, module="conda_build")
+
```

Comment:
We should let the decision on when to display `DeprecationWarning` up to downstream users, i.e., only opt-in to this for our tests and at most at the CLI level (though I can't think of an immediate reason why we'd want this on the CLI).

Reply:
While I agree that we shouldn't override what the downstream user wants, the current configuration means no users of `conda`/`conda-build` get warnings when they use deprecated features (e.g. https://github.com/conda/conda/issues/13704) which is not good

We could wrap this in an `if` block as suggested in the docs: https://docs.python.org/3/library/warnings.html#overriding-the-default-filter

Reply:
We need this for the CLI when CLI arguments have been deprecated

Reply:
See https://github.com/conda/conda-build/pull/5244/commits/f78331dcd88f0b50251044db033230c048cf4005

Reply:
Oh, I didn't realize `DeprecationWarning` was used for CLI args.
IMO, those cases should use `UserWarning` and `DeprecationWarning`/`PendingDeprecationWarning` should only be used when `conda`/`conda_build` is used as a library -- but that'd be something to discuss somewhere else (e.g., the `conda` issue linked above).

----
Having this guarded with `if not sys.warnoptions:` would still mean that downstream (i.e., non-CLI) users don't get full control over it.
E.g., a `conda-smithy` invocation would display the warnings which would not be something one might expect in a usual `DeprecationWarning` workflow.
Meaning, I'd rather put this under `conda_build.cli` rather than general `conda_build` code.
(That all is more of a general design/convention thing; but TBH, it's not as important for `conda-build` since the downstream uses are few/niche and one might argue that it is good to have this verbose for `conda-smithy` since we currently don't have tests for deprecation warnings in its CI.)

Reply:
Reverted in https://github.com/conda/conda-build/pull/5244/commits/29fb7bf588ad55050b34138d410ba7ac21475003

Working on https://github.com/conda/conda/pull/13716 which will need to be ported over to `conda-build`